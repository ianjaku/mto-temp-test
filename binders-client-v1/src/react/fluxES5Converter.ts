/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
