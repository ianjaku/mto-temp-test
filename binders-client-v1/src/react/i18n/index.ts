import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
