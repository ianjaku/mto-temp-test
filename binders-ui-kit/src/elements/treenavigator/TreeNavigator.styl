@import "../../vars.styl";

.tree-navigator
    border: 1px solid borderGrayColor
    width: 100%
    color: baseColor
    font-family: defaultFontName
    font-size: 14px
    user-select: none
    height: 100%
    &-loader
        display: flex
        align-items: center
        justify-content: center
        height: 64px
    &.is-root
        .tree-navigator-parent-back
            display: none

    &-parent,&-row
        height: 44px
        display: flex
        align-items: center
        padding-left: 19px
        &.is-selected
            background-color: accentColor
            color: whiteColor
        &.is-disabled
            color: disabledColor
    &-parent
        border-bottom: 2px solid middleGreyColor
        margin-bottom: 8px
        &-label
            font-weight: 500
        &-back
            margin-right: 12px
            width: 24px
            height: 24px
            border: 1px solid transparent
            border-radius: 5px
            padding-top: 1px
            cursor: pointer
            &:hover
                border-color: baseColor
            &-icon
                /**/
    &-rows
        display: block
        height: calc(100% - 52px)
        overflow-y: auto

        &-empty
            height: 64px
            display: flex
            font-style: italic
            justify-content: center
            align-items: center
            color: middleGreyColor

        &-placeholder
            height: 64px
    &-row
        &-folder-icon
            margin-right: 12px
            margin-left: 3px
        &-label
            /**/
        &-enter
            margin: 0 12px 0 auto
            width: 24px
            height: 24px
            border: 1px solid transparent
            border-radius: 5px
            padding: 1px
            cursor: pointer
            &:hover
                border-color: baseColor

.is-selected
    .tree-navigator-row-enter:hover
        border-color: whiteColor

    .tree-navigator-parent-back:hover
        border-color: whiteColor

