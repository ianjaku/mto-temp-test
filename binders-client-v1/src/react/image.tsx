/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
