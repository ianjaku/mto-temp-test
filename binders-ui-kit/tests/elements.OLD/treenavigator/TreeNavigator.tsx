import * as React from "react";
import TreeNavigator from "../../../src/elements/treenavigator";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const createTreeNavigatorParentItem = () => {
    return [{ id: "0", name: "Root", kind: "document" }];
};

const createTreeNavigatorItems = () => [
    { id: "1", name: "Item 1", kind: "document"},
    { id: "2", name: "Item 2", kind: "document"},
];

let navigatedTo = "-1";
let selected = "-1";
const onNavigate = (id) => navigatedTo = id.id;
const onSelect = (id) => selected = id;

// tslint:disable-next-line
const createTreeNavigator = () => wrapWithTheme(
    <TreeNavigator
        rootItems={createTreeNavigatorParentItem()}
        parentItems={createTreeNavigatorParentItem()}
        items={createTreeNavigatorItems()}
        onNavigate={onNavigate}
        onSelect={onSelect}
    />,
);

describe("Tree navigator", () => {
    test("Tree navigator (snapshot)", () => {
        const treeNavigator = create(createTreeNavigator());
        expect(treeNavigator).toMatchSnapshot();
    });
});

test("Tree navigator fields (mount: enzyme+jsdom)", () => {
    const treeNavigator = mount(createTreeNavigator());
    const parentRow = treeNavigator.find(".tree-navigator-parent").at(0);
    const row1 = treeNavigator.find(".tree-navigator-row").at(0);
    const row2 = treeNavigator.find(".tree-navigator-row").at(1);
    expect(parentRow.find(".tree-navigator-parent-label").first().text()).toEqual("Root");
    expect(row1.find(".tree-navigator-row-label").first().text()).toEqual("Item 1");
    expect(row2.find(".tree-navigator-row-label").first().text()).toEqual("Item 2");
});

test("Tree navigator select (mount: enzyme+jsdom)", () => {
    const treeNavigator = mount(createTreeNavigator());
    const parentRow = treeNavigator.find(".tree-navigator-parent").at(0);
    const row1 = treeNavigator.find(".tree-navigator-row").at(0);
    parentRow.simulate("click");
    expect(selected).toEqual("0");
    row1.simulate("click");
    expect(selected).toEqual("1");
});

test("Tree navigator navigate (mount: enzyme+jsdom)", () => {
    const treeNavigator = mount(createTreeNavigator());
    const row1 = treeNavigator.find(".tree-navigator-row").at(0);
    row1.find(".tree-navigator-row-enter-icon").first().simulate("click");
    expect(navigatedTo).toEqual("1");
    navigatedTo = "-1";
    row1.simulate("doubleclick");
    expect(navigatedTo).toEqual("1");
});
