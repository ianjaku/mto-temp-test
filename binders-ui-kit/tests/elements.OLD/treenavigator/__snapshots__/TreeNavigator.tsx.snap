// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree navigator Tree navigator (snapshot) 1`] = `
<div
  className="tree-navigator is-root"
>
  <div
    className="tree-navigator-parent"
    onClick={[Function]}
    onDoubleClick={[Function]}
  >
    <span
      className="tree-navigator-parent-back"
      onClick={[Function]}
    >
      <span
        className="material-icons tree-navigator-parent-back-icon"
        color="#313131"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#313131",
            "display": "inline-block",
            "fontSize": 20,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        keyboard_arrow_left
      </span>
    </span>
    <label
      className="tree-navigator-parent-label"
    >
      Root
    </label>
  </div>
  <div
    className="tree-navigator-rows"
  >
    <div
      className="tree-navigator-row"
      onClick={[Function]}
      onDoubleClick={[Function]}
    >
      <span
        className="material-icons tree-navigator-row-folder-icon"
        color="#313131"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#313131",
            "display": "inline-block",
            "fontSize": 16,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        folder
      </span>
      <label
        className="tree-navigator-row-label"
      >
        Item 1
      </label>
      <span
        className="tree-navigator-row-enter"
        onClick={[Function]}
      >
        <span
          className="material-icons tree-navigator-row-enter-icon"
          color="#313131"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#313131",
              "display": "inline-block",
              "fontSize": 20,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_right
        </span>
      </span>
    </div>
    <div
      className="tree-navigator-row"
      onClick={[Function]}
      onDoubleClick={[Function]}
    >
      <span
        className="material-icons tree-navigator-row-folder-icon"
        color="#313131"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#313131",
            "display": "inline-block",
            "fontSize": 16,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
          }
        }
      >
        folder
      </span>
      <label
        className="tree-navigator-row-label"
      >
        Item 2
      </label>
      <span
        className="tree-navigator-row-enter"
        onClick={[Function]}
      >
        <span
          className="material-icons tree-navigator-row-enter-icon"
          color="#313131"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#313131",
              "display": "inline-block",
              "fontSize": 20,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_right
        </span>
      </span>
    </div>
  </div>
</div>
`;
